runtime/src/test/java/org/corfudb/runtime/clients/CorfuLayoutHandlerTest.java

```diff
@@ -0,0 +1,235 @@
+import org.corfudb.runtime.view.Layout;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.corfudb.protocols.CorfuProtocolCommon.DEFAULT_UUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLayout.getBootstrapLayoutResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLayout.getCommitLayoutResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLayout.getLayoutResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLayout.getPrepareLayoutResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLayout.getProposeLayoutResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getResponseMsg;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class CorfuLayoutHandlerTest {
+
+    // The BaseHandler instance used for testing
+    private LayoutHandler layoutHandler;
+
+    // Objects that need to be mocked
+    private IClientRouter mockClientRouter;
+    private ChannelHandlerContext mockChannelHandlerContext;
+
+    private final AtomicInteger requestCounter = new AtomicInteger();
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId   indicates if the message is clusterId aware
+     * @param ignoreEpoch       indicates if the message is epoch aware
+     * @return                  the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * Perform the required preparation before running individual
+     * tests by preparing the mocks.
+     */
+    @Before
+    public void setup() {
+        mockClientRouter = mock(IClientRouter.class);
+        mockChannelHandlerContext = mock(ChannelHandlerContext.class);
+        layoutHandler = new LayoutHandler();
+        layoutHandler.setRouter(mockClientRouter);
+    }
+
+    /**
+     * Test that the LayoutHandler correctly handles a LAYOUT_RESPONSE.
+     */
```

Comment:
Would it be worth considering the case where deserializing the Layout throws an exception, and seeing that we complete exceptionally?